This patch addresses linker errors about double defined symbols, which are
caused by defining global variables in a header file that is included by more
than one compilation unit.

--- src/app/tuxmath/src/menu.c
+++ src/app/tuxmath/src/menu.c
@@ -79,7 +79,7 @@
 //const int min_font_size = 8, default_font_size = 20, max_font_size = 33;
 
 /* menu title rect */
-SDL_Rect menu_title_rect;
+extern SDL_Rect menu_title_rect;
 
 /* buffer size used when reading attributes or names */
 const int buf_size = 128;
--- src/app/tuxmath/src/menu_lan.c
+++ src/app/tuxmath/src/menu_lan.c
@@ -37,7 +37,7 @@
 
 
 /* lan_player_type now defined in network.h */
-lan_player_type lan_player_info[MAX_CLIENTS];
+extern lan_player_type lan_player_info[MAX_CLIENTS];
 
 /* Local function prototypes: ------------------- */
 void draw_player_table(void);
--- src/app/tuxmath/src/setup.c
+++ src/app/tuxmath/src/setup.c
@@ -64,7 +64,7 @@
 /* (These are now 'extern'd in "tuxmath.h") */
 
 
-SDL_Surface* screen;
+extern SDL_Surface* screen;
 SDL_Surface* images[NUM_IMAGES];
 sprite* sprites[NUM_SPRITES];
 /* Need special handling to generate flipped versions of images. This
--- src/app/tuxmath/src/titlescreen.c
+++ src/app/tuxmath/src/titlescreen.c
@@ -41,16 +41,16 @@
 int numupdates = 0; // tracks how many blits to be done
 
 // Colors we use:
-SDL_Color black;
+extern SDL_Color black;
 SDL_Color gray;
 SDL_Color dark_blue;
-SDL_Color red;
-SDL_Color white;
-SDL_Color yellow;
+extern SDL_Color red;
+extern SDL_Color white;
+extern SDL_Color yellow;
 SDL_Color bright_green;
 
 // Type needed for trans_wipe():
-struct blit {
+extern struct blit {
     SDL_Surface *src;
     SDL_Rect *srcrect;
     SDL_Rect *dstrect;
--- src/app/tuxmath/src/tuxmath.c
+++ src/app/tuxmath/src/tuxmath.c
@@ -34,6 +34,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 
+extern char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; //!< Global buffer for wrapped lines.
 
 int main(int argc, char* argv[])
 {
--- src/app/tuxmath/t4k/src/t4k_common.h
+++ src/app/tuxmath/t4k/src/t4k_common.h
@@ -239,7 +239,7 @@
 #define MAX_LINES 128     //!< Maximum lines to wrap.
 #define MAX_LINEWIDTH 256 //!< Maximum characters of each line.
 
-char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; //!< Global buffer for wrapped lines.
+extern char wrapped_lines[MAX_LINES][MAX_LINEWIDTH]; //!< Global buffer for wrapped lines.
 
 //TODO separate headers for different areas a la SDL?
 
